fix: file editor not saving when file contents changed

ivyjeong13 opened the pull request:
Addresses #2248 
* when onFileChanged is fired, need to call saveContents

ibuildthecloud commented:
There's a very small nuance that nobody would have known about that needs to be addressed here.  The modified bit and buffer field should only be reset here: https://github.com/obot-platform/obot/blob/c7ac341b83977c96bf652dc4c32af024d590a078/ui/user/src/lib/components/messages/Input.svelte#L59

The basic idea is that the modified flag means that we have modified the file without AI knowing about it.  So on the next chat input we tell the AI that we changed the file.  This isn't perfect, but kind of best effort because various things like refreshing the page will mess it up.  I think the existing `download()` logic was already wrong in that it reset the modified bit.

When tell the LLM that we modified a file we actually send a special json input as the chat input.  You can see the input when you look at the thread in the admin UI.

ivyjeong13 commented:
> There's a very small nuance that nobody would have known about that needs to be addressed here. The modified bit and buffer field should only be reset here:
> 
> https://github.com/obot-platform/obot/blob/c7ac341b83977c96bf652dc4c32af024d590a078/ui/user/src/lib/components/messages/Input.svelte#L59
> 
> The basic idea is that the modified flag means that we have modified the file without AI knowing about it. So on the next chat input we tell the AI that we changed the file. This isn't perfect, but kind of best effort because various things like refreshing the page will mess it up. I think the existing `download()` logic was already wrong in that it reset the modified bit.
> 
> When tell the LLM that we modified a file we actually send a special json input as the chat input. You can see the input when you look at the thread in the admin UI.

Taking a look and will update! Thanks Darren!
